docs/end-user/profiler-fundamentals.md

```diff
@@ -0,0 +1,27 @@
+# Profiler Fundamentals
+
+When profiling code, there are two primary sources of information—stack samples, and markers.
+
+## Samples
+
+For samples, the profiler stops the execution of the profiled code at a fixed rate, for example every 1ms, and records relevant information like the current stack. These samples are aggregated together and provide a statistical look into the execution of the program. There is no guarantee that all of the code that is running will be sampled, but with enough samples, it provides a good look into where time was spent while running the targeted code.
+
+### Statistical significance and the overhead of samples
+
+There are two ways to get more samples to provide more statistical significance. The first is to increase the sampling rate. For example the sampling rate could be increased from 1ms to 0.1ms to collect 10 times more samples. This makes it easy to run the profiled program fewer times, but comes with a trade-off. When the profiler is collecting more samples, the profiler needs to do more work to collect these samples, and thus there is more overhead from the profiler. This can skew the results.
```

Comment:
optional nit: instead of "sampling rate" I'd use "sampling period" as that's what the configured number is (in the 3rd sentence only -- in the 2nd sentence it's fine to keep "rate").

Reply:
Optional: about the overhead, I think there's something about using some hardware support when the period is 1ms or more, which is more efficient. Maybe @mstange could precise this.